<?xml version="1.0" encoding="UTF-8"?>
<launch>

  <param name="robot_description"
    textfile="$(find sensor_driver)/urdf/multivelodyne_3d.urdf" />
    
  <node name="robot_state_publisher" pkg="robot_state_publisher"
    type="robot_state_publisher" />
    
    
  <arg name="rviz" default="true" />
  
  
  <arg name="configfile" default="$(find sensor_driver)/config/confighuachen.xml" />
  <arg name="logdir" default="--log_dir=/home/jkj/catkin_ws/log" />
  <arg name="glogparam" default="--logtostderr=0  --colorlogtostderr=1 --stderrthreshold=5   $(arg logdir)" />

  <node pkg="sensor_driver" type="masternode" name="masternode" args="$(arg glogparam)" output="screen">
  	<param name="config_file" value="$(arg configfile)"/>
  </node>

  
  <node pkg="covgrid_slam_ros" type="lidarodometrymain" name="lidarodometrymain" args="--logtostderr=0  --colorlogtostderr=1 --stderrthreshold=0   $(arg logdir)" output="screen">
  <param name="configuration_directory" value="$(find covgrid_slam_ros)/config" />
  <param name="configuration_basename" value="localmap.lua" />
  <param name="withoutimu" value="true"/>
  <param name="withgps" value="true"/>
  <param name="sendflag" value="false"/>
  <param name="local_ip" value="192.168.0.112"/>
  <param name="local_port" value="9911"/>
  <param name="remote_ip" value="192.168.0.111"/>
  <param name="remote_port" value="9912"/>
  </node>

  <node pkg="covgrid_slam_ros" type="transformmain" name="transformmain" args="$(arg glogparam)" output="screen">
  <param name="local_ip" value="192.168.0.112"/>
  <param name="local_port" value="9906"/>
  <param name="remote_ip" value="192.168.0.111"/>
  <param name="remote_port" value="9905"/>
  <param name="send_flag" value="false"/>
  <param name="borderdetection_flag" value="true"/>
  <param name="heightdiff_threshold" value="0.3"/>
  </node>

  
  
  <node pkg="sensor_driver" type="getmultivelodynedata" name="getmultivelodynedata" args="--logtostderr=0  --colorlogtostderr=1 --stderrthreshold=2   $(arg logdir)" output="screen">
  </node>

  <group if="0">
  <node pkg="sensor_driver" type="getinsdata" name="getinsdata" args="--logtostderr=0  --colorlogtostderr=1 --stderrthreshold=0   $(arg logdir)" output="screen">
  </node>
  </group>
  
  <group if="$(arg rviz)">
    <node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz" args="-d $(find covgrid_slam_ros)/rviz_cfg/covgrid_mapping.rviz" />
  </group>

</launch>
